src/hooks/usePriceEstimation.ts

```diff
@@ -73,7 +104,9 @@ export function usePriceEstimationWithSlippage(params: SlippageParams): Result {
           getPriceParams['amountInUnits'] = amount
         }
 
-        const price = await dexPriceEstimatorApi.getPrice(getPriceParams)
+        const preSlippagePrice = await dexPriceEstimatorApi.getPrice(getPriceParams)
+
+        const price = checkSlippageAgainstPrice(priceSlippage, preSlippagePrice)
```

Comment:
I think we should undo the changes in this file, right?

If any piece needs to apply a slippage, he would get the estimation, and then apply the slippage.

Reply:
done